test/integration/websocket_integration_test.cc

```diff
@@ -60,7 +60,8 @@ void WebsocketIntegrationTest::validateUpgradeRequestHeaders(
     const Http::RequestHeaderMap& original_request_headers) {
   Http::TestRequestHeaderMapImpl proxied_request_headers(original_proxied_request_headers);
   if (proxied_request_headers.ForwardedProto()) {
-    ASSERT_EQ(proxied_request_headers.getForwardedProtoValue(), "http");
+    ASSERT_EQ(proxied_request_headers.getForwardedProtoValue(),
+              downstreamProtocol() == Http::CodecType::HTTP3 ? "https" : "http");
```

Comment:
Hm. This is a bit surprising. Do you have any idea why H2 and H3 are different here?

Reply:
From an (admittedly not crazy long) exploration down said rabbit hole, I think it's because you can run h2 unencrypted (`h2c` _shudders_) but not h3, so the h2 tests took that workaround but not the h3 ones. That also explains why the SNI needed to be fixed in a bunch of places for h3